concourse-webhook-validator/src/concourse_webhook_validator/__init__.py

```diff
@@ -0,0 +1,354 @@
+
+        if webhooks_dict[resource]['dwv'] == 0:
+            delete_webhook_valid = '✅'
+        elif webhooks_dict[resource]['dwv'] == 1:
+            delete_webhook_valid = '❌'
+            resource_validation_failed = 1
+            resource_failed_reasons.append(f"'delete-webhooks' job '{resource}' has a different token to the resource")
+        else:
+            delete_webhook_valid = '❓'
+            resource_validation_failed = 1
+            resource_failed_reasons.append(f"'{resource}' missing 'delete-webhooks' job or webhook token parameter not set")
+        
+        print(f"{resource.ljust(int(resource_name_length) + 1)}|" +
+                f" {resource_webhook_valid.center(int(len(resource_token_valid_text)))}|" +
+                f" {create_webhook_valid.center(int(len(create_token_matches_text)))}|" +
+                f"{delete_webhook_valid.center(int(len(delete_token_matches_text)))}")
+    
+    if resource_validation_failed == 1:
+        print()
+        for reason in resource_failed_reasons:
+            format_output(
+                reason,
+                "warn"
+            )
+    
+    return resource_validation_failed
+
+def main() -> None:
+    parser = argparse.ArgumentParser(description="Program arguments and options")
+    parser.add_argument('pipeline',
+                        default=None,
+                        help="The name of the pipeline to load",
+                        nargs=1
+                        )
+    parser.add_argument('--base-dir',
+                        default='',
+                        help="The base directory in which the pipeline configurations are stored",
+                        nargs=1)
+    parser.add_argument('--deployment',
+                        default='',
+                        help="The name of the Concourse deployment",
+                        nargs=1)
+    parser.add_argument('--team',
+                        default='',
+                        help="The team name that the pipeline is configured for",
+                        nargs=1)
+    args = parser.parse_args()
+    
+    if len(args.base_dir) == 0:
+        pipeline_base_dir = ''
+    else:
+        pipeline_base_dir = args.base_dir[0]
+    
+    if len(args.deployment) == 0:
+        pipeline_deployment = ''
+    else:
+        pipeline_deployment = args.deployment[0]
+    
+    if len(args.team) == 0:
+        pipeline_team = ''
+    else:
+        pipeline_team = args.team[0]
+    
+    pipeline_file = args.pipeline[0]
+    pipeline_file_path = os.path.join(pipeline_base_dir, pipeline_deployment, pipeline_team, pipeline_file)
+
+    webhooks_dict = {}
+    format_output(
+        f"Checking webhooks configuration: {colours.bold}{pipeline_file}{colours.end}"
+    )
+    pipeline_config = load_pipeline_config(pipeline_file_path)
+    if validate_pipeline_config(pipeline_config) == 0:
+        webhooks_dict = parse_pipeline_resources(pipeline_config, webhooks_dict)
```

Comment:
I don't think we have to reassign here. I think we can just pass in directly. I.e. `parse_pipeline_resources(pipeline_config, webhooks_dict)`. As such we can drop the return on the parse_pipeline_resources function

Reply:
The `parse_pipeline_resources` can modify the object and so the reassignment is required to ensure `webhook_dict` is updated.

I could rename the var to make it more explicit if needed.